packages/api-server/src/cache/data.ts

```diff
@@ -153,41 +153,58 @@ export class RedisDataCache {
           // set data cache
           await pubClient.set(dataKey, result, setDataKeyOptions);
           // publish the result to channel
+          const publishResult = successResult(result);
           const totalSubs = await pubClient.publish(
             this.lock.subscribe.channel,
-            result
+            publishResult
           );
           console.debug(
-            `[${this.constructor.name}]: publish message ${result} on channel ${this.lock.subscribe.channel}, total subscribers: ${totalSubs}`
+            `[${this.constructor.name}][success]: publish message ${publishResult} on channel ${this.lock.subscribe.channel}, total subscribers: ${totalSubs}`
           );
           await releaseLock(lockValue);
           return result;
-        } catch (error) {
-          console.debug(error);
+        } catch (error: any) {
+          const reason = JSON.stringify(error.message);
+          if (!reason.includes("request to")) {
+            // publish the non-network-connecting-error-result to channel
+            const publishResult = errorResult(reason);
+            const totalSubs = await pubClient.publish(
+              this.lock.subscribe.channel,
+              publishResult
+            );
+            console.debug(
+              `[${this.constructor.name}][error]: publish message ${publishResult} on channel ${this.lock.subscribe.channel}, total subscribers: ${totalSubs}`
+            );
+          }
           await releaseLock(lockValue);
+          throw new Error(error.message);
         }
       }
 
       // if lock is not acquired
       try {
-        const msg = await this.subscribe();
+        const result = await this.subscribe();
         console.debug(
           `[${this.constructor.name}]: hit cache via Redis.Subscribe, key: ${dataKey}`
         );
-        return msg;
+        return result;
       } catch (error: any) {
         if (
           !JSON.stringify(error).includes(
             "subscribe channel for message time out"
           )
         ) {
-          console.debug(`[${this.constructor.name}]: subscribe err:`, error);
+          console.debug(
+            `[${this.constructor.name}]: subscribe err:`,
+            error.message
+          );
+          throw new Error(error.message);
```

Comment:
if we throw error here, it seems that the `check if poll time out` logic won't have chance to run.

Reply:
> if we throw error here, it seems that the `check if poll time out` logic won't have chance to run.

[here](https://github.com/nervosnetwork/godwoken-web3/pull/173/files/5337c500dc9e9b9414b2b895095c9367996a60c5#diff-03bc3b7de02c3fd410085215655392a45a3696958610bde9668514dbda132289R193) when one subscribtion timeout, it will keep polling. the reason we throw error here is to let the rest client subscribed to our channel knows that right now the executing call went wrong and the wrong result can be cache too. so that we don't need to repeat executing all the wrong call again and again. the deed loop actually is coming from not cacheing the wrong result. so the rest of client just keep waitting.

Reply:
> Shall we tune the default `POLL_TIME_OUT_MS` smaller? Maybe smaller than 100s? Maybe 90s? And we should keep tracking the slowest request leading us to the better TIME_OUT value.

sure we can set a more unique `TIME_OUT` value, the original value is [2 minutes](https://github.com/nervosnetwork/godwoken-web3/blob/main/packages/api-server/src/cache/data.ts#L24), I think 100s might comes from some other connection timeout limitation etc, not from here. it did not trigger `POLL_TIME_OUT_MS` actually.

Reply:
> [here](https://github.com/nervosnetwork/godwoken-web3/pull/173/files/5337c500dc9e9b9414b2b895095c9367996a60c5#diff-03bc3b7de02c3fd410085215655392a45a3696958610bde9668514dbda132289R193) when subscribe timeout, it will keep polling until polling time out. the reason we throw error here is to let the rest clients subscribed to our channel knows that right now the executing call went wrong and the wrong result can be cache too. so that we don't need to repeat executing all the wrong call again and again. the deed loop actually is coming from not cacheing the wrong result. so the rest of client just keep waitting.

yes, you are right.